docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -957,6 +957,27 @@ Alternatively, you can use the `@LookupIfProperty` and `@LookupUnlessProperty` a
  }
 ----
 
+if there is more than one bean that matches the required type and qualifiers and is eligible for injection, it is possible to iterate (or stream) available bean instances.
```

Comment:
I think that this paragraph would deserve a separate section with title along the lines of _"Sorting beans obtained with programmatic lookup"_.

Reply:
I agree, I'd add it as a separate non-standard feature.
Ideally with a use case (why you might need it) and with an example showing two prioritized beans (and third with no prio?) and what the ordering looks like.

Reply:
@mkouba , just updated the PR, WDYT ?